elastic/kibana
# [Cloud Security] CSPM AWS agentless selector

## Summary

A part of 
- https://github.com/elastic/security-team/issues/7969

This PR doesn't cover the fact that with Agentless only `Manual - Direct Access Key` and `Manual - Temprary access key` are supported. Hiding other setup options, like `Cloudformation`, `Manual - Assume role` or Manual - Shared Credentials`, will be resolved in the next PR. As this feature is hidden behind a feature flag and based on the fact that Agentless is available only for a handful of internal Cloud org ids, it should be safe to merge this PR if code-wise it's ok.

### How to test

Make sure to have the FF in your `serverless.security.dev.yml` (it's basically the `kibana.dev.yml` but specifically for Serverless Security Projects) enabled. Also specify some serverless project id, to enable the logic of `isServerlessEnabled`
```
xpack.fleet.enableExperimental: ['agentless']
xpack.cloud.serverless.project_id: 'some_fake_project_id'
``` 

Follow the steps from this comment https://github.com/elastic/security-team/issues/7972#issuecomment-1808096052 to have the Agentless artifacts (agent, policy, output, and fleet server host) locally

After that, you should be able to test the flow. Mind that with agentless only `Manual - Direct Access Key` and `Manual - Temprary access key` setup methods are available. Hiding other options will be addressed in the follow ups

### Screencast

[screencast-maxcold-agentless-selector-pr-b5b706.kb.eu-west-1.aws.qa.elastic.cloud-2023.11.27-10_30_57.webm](https://github.com/elastic/kibana/assets/478762/b3abd216-0863-4081-8643-cda0688757f6)


### Checklist
